### PR TITLE
nmssm condor analysis

### PR DESCRIPTION
**Pull request by tvoigtlaender**
Branch of nmssm_analysis_68trainings.

Executes the tensorflow training on the condor cluster using its GPUs.
The main script run_ml_condor.sh calls the parts of the run as run_ml.sh did.
It only executes the missing parts of the script (no training if models are already in the output).
It also allows partial execution (only training, only testing, etc.) by adding an option.
Instead of calling ml/run_training.sh, run_ml_condor.sh calls the new scripts in ml_condor.
These scripts take the needed datasets, set up a job for condor for the training
and send the results back to the matching output directory.

Also fixed a missing part of ml/run_testing.sh

### CODE REVIEW

File: ml_condor/run_condor_training_gpu.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+# Steps 2 and 3 are functionally identical to ml/run_training.sh
+
+set -e
+ERA=$1
+CHANNEL=$2
+TAG=$3
+folder=${ERA}_${CHANNEL}_${TAG}
+outdir=/tmp/${folder}
+#---1---
+cephdir=root://ceph-node-a.etp.kit.edu:1094//tvoigtlaender/nmssm_data
```
Comment (harrypuuter) [inline ml_condor/run_condor_training_gpu.sh]: make this user independent

File: ml_condor/run_condor_training_gpu.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+if [[ $ERA == "all_eras" ]]; then
+  ERAS="2016 2017 2018"
+else
+ ERAS=$ERA
+fi
+
+echo $ERAS
+
+# Copy the needed datasets from /ceph
+for ERA_LOOP in ${ERAS}; do
```
Comment (harrypuuter) [inline ml_condor/run_condor_training_gpu.sh]: ERA instead of ERA_LOOP makes more sence semantically

Comment (harrypuuter) [reply ml_condor/run_condor_training_gpu.sh]: same for folder and outdir variables

File: run_ml_condor.sh [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+RECALC=$5 # recalculate certain stages of the programm: "a" datasets, "b" training, "c" testing. Can be combined (ac)
+
+#for MASS in 240 280 320 360 400 450 500 550 600 700 800 900 1000 1200 
+#do
+#for BATCH in `python ml/get_nBatches.py ${MASS}`
+#do
+
+SET=${ERA}_${CHANNEL}_${MASS}_${BATCH}
+OUTPUT_PATH=output/ml/${SET}
+CEPH_PATH=/ceph/srv/tvoigtlaender/nmssm_data/${SET}
```
Comment (harrypuuter) [inline run_ml_condor.sh]: as above

File: run_ml_condor.sh [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  mkdir ${CEPH_PATH}
+fi
+if [[ ! -d output/log/logandrun ]]; then
+  echo create output/log/logandrun
+  mkdir -p output/log/logandrun
+fi
+#Test if needed trainings data is on ceph
+CEPH_TEST=false
+for ERA_LOOP in ERAS; do
+  CEPH_PATH_LOOP=/ceph/srv/tvoigtlaender/nmssm_data/${ERA_LOOP}_${CHANNEL}_${MASS}_${BATCH}
```
Comment (harrypuuter) [inline run_ml_condor.sh]: as above

File: run_ml_condor.sh [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      wait
+      ./ml/combine_configs.sh all_eras $CHANNEL ${MASS}_${BATCH}
+    else
+      ./ml/create_training_dataset.sh $ERA $CHANNEL $MASS $BATCH
+    fi
+    # If dataset creation specified:
+    if [[ ${RECALC} == *"a"* ]]; then
+      # Remove data on ceph
+      for ERA_LOOP in ${ERAS}; do
+        CEPH_PATH_LOOP=/ceph/srv/tvoigtlaender/nmssm_data/${ERA_LOOP}_${CHANNEL}_${MASS}_${BATCH}
```
Comment (harrypuuter) [inline run_ml_condor.sh]: as above

File: run_ml_condor.sh [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        rm ${CEPH_PATH_LOOP}/*
+      done
+    fi
+  fi
+  # Upload dataset to ceph
+  for ERA_LOOP in ${ERAS}; do
+    SET_LOOP=${ERA_LOOP}_${CHANNEL}_${MASS}_${BATCH}
+    OUTPUT_PATH_LOOP=output/ml/${SET_LOOP}
+    CEPH_PATH_LOOP=/ceph/srv/tvoigtlaender/nmssm_data/${SET_LOOP}
```
Comment (harrypuuter) [inline run_ml_condor.sh]: as above

**Review by harrypuuter — CHANGES_REQUESTED** [review]
some small comments, other than that looks good to me !

File: ml_condor/run_condor_training_gpu.sh [hunk 7]
```diff
@@ -10,14 +10,13 @@
 # Steps 2 and 3 are functionally identical to ml/run_training.sh
 
 set -e
-ERA=$1
+ERA_NAME=$1
 CHANNEL=$2
 TAG=$3
-folder=${ERA}_${CHANNEL}_${TAG}
-outdir=/tmp/${folder}
+NAME_USER=$4
```
Comment (harrypuuter) [inline ml_condor/run_condor_training_gpu.sh:19]: you could simply use `${USER}`, this should be your username, so you don't need to add it as an argument

**Review by harrypuuter — CHANGES_REQUESTED** [review]

File: ml_condor/run_condor_training_gpu.sh (hunk 7 above)
Comment (tvoigtlaender) [reply ml_condor/run_condor_training_gpu.sh:19]: I don't believe ${USER} can be used here as this part of the code is executed in a docker image.

**Pull request closed without merging**